### PR TITLE
kafka-to-kv template

### PR DESCRIPTION
**Pull request by sahare92**

### CODE REVIEW

File: kafka-to-kv/function.yaml.values [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  description: "Kafka topic name"
+  required: true
+
+Broker:
+  displayName: Kafka URL and port
+  kind: string
+  description: "Kafka broker address"
+  required: true
+
+NginxHost:
```
Comment (yaronha) [inline kafka-to-kv/function.yaml.values]: @sahare92 need to change all mentions of `nginx` to `V3ioApiEndpoint` (nginx is an internal and very unclear term)

File: kafka-to-kv/kafka-to-kv.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+def generate_payload(event_body):
+    msg = json.loads(event_body)
+    attributes = generate_attributes(msg, EVENT_KEY)
+    payload = {}
+    payload = insert_key(payload, msg[EVENT_KEY])
+    payload = insert_attributes(payload, attributes)
+    return payload
+
+
+def handler(context, event):
```
Comment (pavius) [inline kafka-to-kv/kafka-to-kv.py]: Handler should be first function, followed by init_context (here you don't have it, but if you do that's the convention).

All other functions should start with an underscore

File: kafka-to-kv/kafka-to-kv.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+NGINX_PORT = os.environ['NGINX_PORT']
+TABLE_NAME = os.environ['TABLE_NAME']
+CONTAINER_NAME = os.environ['CONTAINER_NAME']
+EVENT_KEY = os.environ['EVENT_KEY']
+USERNAME = os.environ['USERNAME']
+PASSWORD = os.environ['PASSWORD']
+
+def get_request_url():
+
+    return 'http://{0}:{1}/{2}/{3}/'.format(NGINX_HOST, NGINX_PORT, CONTAINER_NAME, TABLE_NAME)
```
Comment (pavius) [inline kafka-to-kv/kafka-to-kv.py]: Use Python3 formatted strings `f'http://{NGINX_HOST}:{NGINX_PORT}/{CONTAINER_NAME}/{TABLE_NAME}/'` and remove newline

File: kafka-to-kv/kafka-to-kv.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+           if type(json_object[key]) == int:
+               attributes[key] = {"N": str(json_object[key])}
+           if type(json_object[key]) == float:
+               attributes[key] = {"N": str(json_object[key])}
+           if type(json_object[key]) == str:
+               attributes[key] = {"S": json_object[key]}
+   return attributes
+
+
+def generate_payload(event_body):
```
Comment (pavius) [inline kafka-to-kv/kafka-to-kv.py]: 1. AFAIK you should have done `event_body.decode('utf-8')`for this not to raise
2. No need to set the first payload, just pass an empty dict to insert_key
3. Just return from `insert_attributes`

File: kafka-to-kv/kafka-to-kv.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            if not item['Item'][attr]:
+                item['Item'].pop(attr)
+    return item
+
+
+def generate_attributes(json_object, object_key):
+   attributes = {}
+   for key in json_object:
+       if key != object_key:
+           if type(json_object[key]) == bool:
```
Comment (pavius) [inline kafka-to-kv/kafka-to-kv.py]: 1. get type once, compare multiple times
2. use `is` not `==`
3. use else if or a lookup dict, you're just comparing after the match

File: kafka-to-kv/kafka-to-kv.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+EVENT_KEY = os.environ['EVENT_KEY']
+USERNAME = os.environ['USERNAME']
+PASSWORD = os.environ['PASSWORD']
+
+def get_request_url():
+
+    return 'http://{0}:{1}/{2}/{3}/'.format(NGINX_HOST, NGINX_PORT, CONTAINER_NAME, TABLE_NAME)
+
+
+def create_encoded_auth():
```
Comment (pavius) [inline kafka-to-kv/kafka-to-kv.py]: Let's simplify and use an off the shelf package for this. requests can also do with `HTTPBasicAuth`

**Pull request merged**